src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java

```diff
@@ -158,6 +160,50 @@
         return flowService.patchFlow(flowId, flowPatchDto);
     }
 
+    /**
+     * Get existing flow loops.
+     *
+     * @param flowId filter by flow id
+     * @param switchId filter by switch id
+     * @return list of flow loops
+     */
+    @ApiOperation(value = "Get flow loops", response = FlowLoopResponse.class, responseContainer = "List")
+    @GetMapping(value = "/loops")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<List<FlowLoopResponse>> getFlowLoops(
+            @RequestParam(value = "flow_id", required = false) String flowId,
+            @RequestParam(value = "switch_id", required = false) String switchId) {
+        return flowService.getFlowLoops(flowId, switchId);
+    }
+
+    /**
+     * Create flow loop.
+     *
+     * @param flowId flow id
+     * @param flowLoopPayload parameters for flow loop
+     * @return created flow loop
+     */
+    @ApiOperation(value = "Create flow loop", response = FlowLoopResponse.class)
+    @PostMapping(value = "/{flow_id}/loops")
```

Comment:
Are we going to have more than 1 loop per flow?

Reply:
No, just a naming convention. We have plural forms everywhere.